modules/sdp/SdpSimulcast.ts

```diff
@@ -0,0 +1,307 @@
+
+        if (simGroup) {
+            return simGroup.ssrcs.split(' ').map(ssrc => parseInt(ssrc));
+        }
+
+        return [ parseInt(mLine.ssrcs?.[0]?.id?.toString()) ];
+    }
+
+    /**
+     * Munges the given media description to enable simulcast for the video media sections that are in either have
+     * SENDRECV or SENDONLY as the media direction thereby ignoring all the RECVONLY transceivers created for remote
+     * endpoints.
+     * NOTE: This needs to be called only when simulcast is enabled.
+     *
+     * @param description
+     * @returns
+     */
+    mungeLocalDescription(description: Description) : Description {
+        if (!(description && description?.sdp !== '')) {
+            return description;
+        }
+        const session = transform.parse(description.sdp);
+
+        for (let media of session.media) {
+            // Ignore recvonly and inactive transceivers created for remote sources.
+            if (media.direction === MediaDirection.RECVONLY || media.direction === MediaDirection.INACTIVE) {
+                continue;
+            }
+
+            // Ignore audio m-lines.
+            if (media.type !== MediaType.VIDEO) {
+                continue;
+            }
+            const mid = media.mid;
+            const numSsrcs = new Set(media.ssrcs?.map(ssrcInfo => ssrcInfo.id));
+            const numGroups = media.ssrcGroups?.length ?? 0;
+            let primarySsrc: number;
+
+            // Do not munge if the description has no ssrcs or if simulcast is already enabled.
+            if (numSsrcs.size === 0 || numSsrcs.size > 2 || (numSsrcs.size === 2 && numGroups === 0)) {
+                continue;
+            }
+            if (numSsrcs.size === 1) {
+                primarySsrc = parseInt((media.ssrcs[0]?.id).toString());
+            } else {
+                const fidGroup = media.ssrcGroups.filter(group => group.semantics === 'FID')?.[0];
+
+                primarySsrc = parseInt(fidGroup.ssrcs.split(' ')[0]);
+            }
+
+            if (this._ssrcCache.has(mid)) {
+                media = this._fillSsrcsFromCache(media);
+            } else {
+                media = this._generateNewSsrcsForSimulcast(media, primarySsrc);
+                const simulcastSsrcs = this._parseSimLayers(media);
+
+                // Update the SSRCs in the cache so that they can re-used for the same mid again.
+                this._ssrcCache.set(mid, simulcastSsrcs);
+            }
+        }
+
+        return new RTCSessionDescription({
+            type: description.type,
+            sdp: transform.write(session)
+        });
+    }
+
+    /**
+     * Munges the given media description by removing the SSRCs and related FID groups for the higher layer streams.
+     *
+     * @param description
+     * @returns
+     */
+    mungeRemoteDescription(description: Description) : Description {
+        if (!(description && description?.sdp !== '')) {
```

Comment:
nit: the negated condition is a little bit hard to read. How about `if (!description || !description.sdp)`  ? This will account for `sdp` being `undefined` or empty.